client/app/pages/dashboards/DashboardPage.jsx

```diff
@@ -144,36 +144,48 @@ DashboardComponent.propTypes = {
   dashboard: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
 };
 
-function DashboardPage({ dashboardSlug, onError }) {
+function DashboardPage({ dashboardSlug, dashboardId, onError }) {
   const [dashboard, setDashboard] = useState(null);
   const onErrorRef = useRef();
   onErrorRef.current = onError;
 
   useEffect(() => {
-    Dashboard.get({ slug: dashboardSlug })
+    Dashboard.get({ id: dashboardId, slug: dashboardSlug })
       .then(dashboardData => {
         recordEvent("view", "dashboard", dashboardData.id);
         setDashboard(dashboardData);
       })
       .catch(error => onErrorRef.current(error));
-  }, [dashboardSlug]);
+  }, [dashboardSlug, dashboardId]);
 
   return <div className="dashboard-page">{dashboard && <DashboardComponent dashboard={dashboard} />}</div>;
 }
 
 DashboardPage.propTypes = {
-  dashboardSlug: PropTypes.string.isRequired,
+  dashboardSlug: PropTypes.string,
+  dashboardId: PropTypes.string,
   onError: PropTypes.func,
 };
 
 DashboardPage.defaultProps = {
+  dashboardSlug: null,
+  dashboardId: null,
   onError: PropTypes.func,
 };
 
+// route kept for backward compatibility
 routes.register(
   "Dashboards.ViewOrEdit",
   routeWithUserSession({
     path: "/dashboard/:dashboardSlug",
     render: pageProps => <DashboardPage {...pageProps} />,
   })
 );
+
+routes.register(
+  "Dashboards.ViewOrEditWithId",
+  routeWithUserSession({
+    path: "/dashboards/:dashboardId-(.*)?",
```

Comment:
If `- ` is optional - probably it should be inside of brackets as well

Reply:
I thought so, I tried it in many ways, but this is the only way I got it to work as I wanted 😅

Reply:
:thinking:

Reply:
Worth adding a Cypress test that tries to load dashboard with `id`, `id-`, `id-slug`, `id-random string`.

Reply:
When you move `-` to brackets it decides that `(-.*)?` is a pattern for `dashboardId`. This should work: `/dashboards/:dashboardId([^-]+)(-.*)?`

Reply:
> Worth adding a Cypress test that tries to load dashboard with id, id-, id-slug, id-random string.

Yep, that's actually on my plans :)

Reply:
> When you move - to brackets it decides that (-.*)? is a pattern for dashboardId.

Oh, I see, it's the same pattern for both indicating I want ath and deciding the attr pattern 🤦‍♂️... Thanks! :)